src/frontend/js/widgets/SyllabusCourseRunsList/components/CourseProductItem/index.tsx

```diff
@@ -147,20 +146,31 @@ const Content = ({ product, order }: { product: Joanie.Product; order?: Joanie.O
   );
 };
 
-const CourseProductItem = ({ productId, courseCode, compact = false }: Props) => {
+const CourseProductItem = ({ productId, courseCode, compact = false }: CourseProductItemProps) => {
   const { item: courseProductRelation, states: productQueryStates } = useCourseProduct(courseCode, {
     productId,
   });
   const product = courseProductRelation?.product;
-  const { item: order, states: orderQueryStates } = useProductOrder({ productId, courseCode });
+  const { item: productOrder, states: orderQueryStates } = useProductOrder({
+    productId,
+    courseCode,
+  });
 
+  // FIXME(rlecellier): useCourseProduct need's a filter on product.type that only return
+  // CredentialOrder
+  const order = productOrder as CredentialOrder;
   const isPendingState = !order || order.state === OrderState.PENDING;
   const hasPurchased = (order && order.state === OrderState.VALIDATED) ?? false;
 
   const hasError = Boolean(productQueryStates.error);
   const isFetching = productQueryStates.fetching || orderQueryStates.fetching;
   const canShowContent = !compact || hasPurchased;
 
+  if (product && product.type !== ProductType.CREDENTIAL) {
+    // CourseProductItem only handle CREDENTIAL products
+    return null;
+  }
+
```

Comment:
I'll replace this by: 
```ts
useMemo(() => {
  if (product && product.type !== ProductType.CREDENTIAL) {
    handle(new Error('Synchronized certificate product course run should not be open.'));
  }
}, [product]);
```

Reply:
According to our discussion, I'll integrate this check to set the value of `hasError`.

Then, I'll handle the error within a useEffect.

```ts
const hasError = Boolean(productQueryStates.error || (product && product.type !== ProductType.CREDENTIAL))

...

useEffect(() => {
  if (product && product.type !== ProductType.CREDENTIAL) {
    handle(new Error('Synchronized certificate product course run should not be open.'));
  }
}, [product])
```